junit5/src/main/java/org/jboss/weld/junit5/auto/ClassScanning.java

```diff
@@ -320,4 +321,13 @@ private static boolean isFieldShadowedBy(Field upper, Field lower) {
         return findFirstAnnotatedConstructor(clazz.getSuperclass(), annotationType);
     }
 
+    private static Class<?> unwrapInstanceTypeParameter(Field field) {
+        Class<?> type = field.getType();
+        if (type.equals(Instance.class)) {
+            ParameterizedType parameterizedType = (ParameterizedType) field.getGenericType();
```

Comment:
Hm, is this going to work if the bean type itself is parameterized? E.g. it can look something like this - `Foo<String>`

Reply:
This is just a minor thing though, the current proposed solution is surely still better than what we had :-)

Reply:
good point with the parameterized type parameter.